# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationUtils.h
# Commit: 1c6ace5aa850
# Full Hash: 1c6ace5aa8502311912112a4a8d80739253579a2
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-07-16 21:45:35
# Regressor Bug: 1972117
# File Overlap Count: 1
# Description:
#   Bug 1972117 - Part 3: Move NotificationsCallback to NotificationUtils r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D255860
# ==============================================================================

diff -r 5db864f2389f -r 1c6ace5aa850 dom/notification/NotificationUtils.h
--- a/dom/notification/NotificationUtils.h	Wed Jul 16 15:33:00 2025 +0000
+++ b/dom/notification/NotificationUtils.h	Wed Jul 16 15:33:00 2025 +0000
@@ -69,6 +69,12 @@
 
 nsCOMPtr<nsINotificationStorage> GetNotificationStorage(bool isPrivate);
 
+using NotificationsPromise =
+    MozPromise<CopyableTArray<IPCNotification>, nsresult, false>;
+
+already_AddRefed<NotificationsPromise> GetStoredNotificationsForScope(
+    nsIPrincipal* aPrincipal, const nsACString& aScope, const nsAString& aTag);
+
 nsresult GetOrigin(nsIPrincipal* aPrincipal, nsString& aOrigin);
 
 nsresult PersistNotification(nsIPrincipal* aPrincipal,